mapper/orm/src/main/java/org/hibernate/search/mapper/orm/loading/impl/HibernateOrmEntityIdEntityLoadingStrategy.java

```diff
@@ -147,25 +153,14 @@ private PojoSelectionEntityLoader<?> doCreate(EntityMappingType entityMappingTyp
 		// We must pass rootEntityMappingType here, to avoid getting a WrongClassException when loading from the cache,
 		// even if we know we actually want instances from the most specific entity type,
 		// because that exception cannot be recovered from.
-		return new HibernateOrmSelectionEntityByIdLoader<>( rootEntityMappingType, queryFactory, sessionContext,
+		return new HibernateOrmSelectionEntityByIdLoader<>( rootEntityMappingType,
+				TypeQueryFactory.create(
+						session.getSessionFactory(), entityMappingType,
+						entityMappingType.getIdentifierMapping().getAttributeName() ),
```

Comment:
```suggestion
				createFactory( entityMappingType ),
```

Also, I'm confused as to why you don't need the same change in `HibernateOrmNonEntityIdPropertyEntityLoadingStrategy.java`?

Also, does this mean you're also fixing a problem for entity loading during search? That would be awesome.

Reply:
> Also, I'm confused as to why you don't need the same change in HibernateOrmNonEntityIdPropertyEntityLoadingStrategy.java?

I've probably just missed it since I was mostly looking at the entity loading query 🙈 

> Also, does this mean you're also fixing a problem for entity loading during search? That would be awesome.

I think it should; since I had some tests failing related to entity projection loading while making changes here, but I'll doublecheck 😃